src/main/java/com/google/cloud/tools/libraries/MavenCoordinates.java

```diff
@@ -0,0 +1,209 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.libraries;
+
+import com.google.common.base.Preconditions;
+import java.text.MessageFormat;
+
+/**
+ * Describes a Maven artifact.
+ */
+public class MavenCoordinates {
+
+  public static final String LATEST_VERSION = "LATEST";
+  public static final String JAR_TYPE = "jar";
+  public static final String MAVEN_CENTRAL_REPO = "central";
+
+  private String repository = MAVEN_CENTRAL_REPO;
+  private String groupId;
+  private String artifactId;
+  private String version = LATEST_VERSION;
+  private String type = JAR_TYPE;
+  private String classifier;
+
+  /**
+   * @param groupId the Maven group ID, cannot be <code>null</code>
+   * @param artifactId the Maven artifact ID, cannot be <code>null</code>
+   */
+  public MavenCoordinates(String groupId, String artifactId) {
+    Preconditions.checkNotNull(groupId, "groupId null");
+    Preconditions.checkNotNull(artifactId, "artifactId null");
+    Preconditions.checkArgument(!groupId.isEmpty(), "groupId empty");
+    Preconditions.checkArgument(!artifactId.isEmpty(), "artifactId empty");
+
+    this.groupId = groupId;
+    this.artifactId = artifactId;
+  }
+
+  /**
+   * @return the Maven version of the artifact, defaults to special value
+   *         {@link MavenCoordinates#LATEST_VERSION}, never <code>null</code>
+   */
+  public String getVersion() {
+    return version;
+  }
+
+  /**
+   * @return the Maven packaging type, defaults to <code>jar</code>, never <code>null</code>
+   */
+  public String getType() {
+    return type;
+  }
+
+  /**
+   * @return the Maven classifier or <code>null</code> if it was not set
+   */
+  public String getClassifier() {
+    return classifier;
+  }
+
+  /**
+   * @return the URI or the identifier of the repository used to download the artifact from, never
+   *         <code>null</code>
+   */
+  public String getRepository() {
+    return repository;
+  }
+
+  /**
+   * @return the Maven group ID, never <code>null</code>
+   */
+  public String getGroupId() {
+    return groupId;
+  }
+
+  /**
+   * @return the Maven artifact ID, never <code>null</code>
+   */
+  public String getArtifactId() {
+    return artifactId;
+  }
+
+  @Override
+  public String toString() {
+    return MessageFormat.format("MavenCoordinates [repository={0}, {1}:{2}:{3}:{4}:{5}]",
+                                repository, groupId, artifactId, type, classifier, version);
+  }
+  
+  @Override 
+  public int hashCode() {
+    return (groupId + ":" + artifactId + ":" + version).hashCode();
+  }
+  
+  @Override 
+  public boolean equals(Object o) {
+    if (o == null || !(o instanceof MavenCoordinates)) {
+      return false;
+    }
+    MavenCoordinates other = (MavenCoordinates) o;
```

Comment:
`equals()` needs to check `repository`, `classifier`, and `type`.   Use `Object.equals()`?

Reply:
The classifier and type are pretty key coordinates.  I think you should have a different method for identifying equivalence that ignores repository. Or perhaps the repository should not be a coordinate at all.

Reply:
Now that you mention it, what I really need to verify here is that the groupId:artifactId is unique, not even including version.